src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/CryptoHelper.cs

```diff
@@ -0,0 +1,442 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Security.Cryptography;
+using CoreWCF.IdentityModel.Tokens;
+using Linq = System.Linq;
+
+namespace CoreWCF.IdentityModel
+{
+    internal static class CryptoHelper
+    {
+        private static RandomNumberGenerator s_random;
+        private const string SHAString = "SHA";
+        private const string SHA1String = "SHA1";
+        private const string SHA256String = "SHA256";
+        private const string SystemSecurityCryptographySha1String = "System.Security.Cryptography.SHA1";
+        static byte[] emptyBuffer;
+
+        private static Dictionary<string, Func<object>> s_algorithmDelegateDictionary = new Dictionary<string, Func<object>>();
+        private static object s_algorithmDictionaryLock = new object();
+
+        internal static byte[] EmptyBuffer
+        {
+            get
+            {
+                if (emptyBuffer == null)
+                {
+                    byte[] tmp = new byte[0];
+                    emptyBuffer = tmp;
+                }
+                return emptyBuffer;
+            }
+        }
+
+        public static bool FixedTimeEquals(byte[] a, byte[] b)
+        {
+            if (a == null && b == null)
+            {
+                return true;
+            }
+            else if (a == null || b == null)
+            {
+                return false;
+            }
+            else if (a.Length != b.Length)
+            {
+                return false;
+            }
+
+            var result = 0;
+            var length = a.Length;
+
+            for (int i = 0; i < length; i++)
+            {
+                result |= a[i] ^ b[i];
+            }
+
+            return result == 0;
+        }
+
+        internal static bool IsSymmetricAlgorithm(string algorithm)
```

Comment:
I was curious about how we can get away with this not being supported so I looked into how it's used on .NET Framework and it's only used by implementations of the abstract method SecurityKey.IsSymmetricAlgorithm. Nothing in-box calls that method. The same goes for IsAsymmetricAlgorithm. So can you remove those two abstract methods from SecurityKey, then remove this method and IsAsymmetricAlgorithm further down.